src/pages/certification/guide.tsx

```diff
@@ -0,0 +1,127 @@
+import Image from 'next/image';
+import { useRouter } from 'next/router';
+import React, { useEffect, useRef, useState } from 'react';
+import styled from 'styled-components';
+
+import Header from '@components/Header';
+import Button from '@components/Shared/Button';
+import convertFileToBlobUrl from '@utils/convertFileToBlobUrl';
+
+const CertificationGuide = () => {
+  const { query, push } = useRouter();
+  const { image, location, name } = query;
+  const [receiptFile, setReceiptFile] = useState<File | null>(null);
+  const fileInputRef = useRef<HTMLInputElement>(null);
+
+  const handleOnClickButton = () => fileInputRef?.current?.click();
+
+  const handleOnChangeFileInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const {
+      target: { files },
+    } = e;
+
+    if (!files?.length) return;
+
+    setReceiptFile(files[0]);
+  };
+
+  useEffect(() => {
+    const getPreviewReceipt = async (file: File) => await convertFileToBlobUrl(file);
+
+    if (receiptFile) {
+      getPreviewReceipt(receiptFile).then((blobUrl) =>
+        push({ pathname: '/certification', query: { blobUrl, image, location, name } }),
+      );
+    }
+  }, [push, receiptFile, image, location, name]);
```

Comment:
의존성 배열이 다소 커보여 생각해 본건데 혹시 이렇게 하는건 무리가 있을까요??

```suggestion
  useEffect(() => {
    const getPreviewReceipt = async (file: File) => await convertFileToBlobUrl(file);
    const { image, location, name } = query;

    if (receiptFile) {
      getPreviewReceipt(receiptFile).then((blobUrl) =>
        push({ pathname: '/certification', query: { blobUrl, image, location, name } }),
      );
    }
  }, [query, push, receiptFile]);
```

Reply:
굿굿 의존성은 제안주신대로 바꾸게씁니다~~
근데 `FileReader`는 더 이상 안 쓰는 방식으로 변경되었어용!